### PR TITLE
Lt 1006 raise a clc

### PR DESCRIPTION
**Pull request by 5pence**
please review

### CODE REVIEW

File: goods/views.py [hunk 1]
```diff
@@ -27,14 +31,33 @@ def get(self, request):
...
     def post(self, request):
         organisation = get_organisation_by_user(request.user)
-
         data = JSONParser().parse(request)
         data['organisation'] = organisation.id
         data['status'] = GoodStatus.DRAFT
+        # if data['not_sure_details_control_code'] is not None:
+        #     data['control_code'] = data['not_sure_details_control_code']
         serializer = GoodSerializer(data=data)
+        #print(data['validate_only'])
```
Comment (markm5A8Z9F43913A) [inline goods/views.py]: Are these commented out lines needed at all?

File: case_types/tests.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (jfaracik) [inline case_types/tests.py]: Delete if empty

File: case_types/views.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.shortcuts import render
+
+# Create your views here.
```
Comment (jfaracik) [inline case_types/views.py]: Delete if empty

File: case_types/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Generated by Django 2.2.2 on 2019-07-01 11:21
+
+from django.db import migrations, models
+import uuid
+
+
+def initialize(apps, schema_editor):
```
Comment (jfaracik) [inline case_types/migrations/0001_initial.py]: Should this be a table as opposed to a choice field?

File: cases/models.py [hunk 5]
```diff
@@ -14,7 +16,12 @@ class Case(models.Model):
     Wrapper for application model intended for internal users.
     """
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    application = models.ForeignKey(Application, related_name='case', on_delete=models.CASCADE)
+    case_type = models.ForeignKey(CaseType,
+                                  related_name='case',
+                                  on_delete=models.DO_NOTHING,
+                                  default='0ec51727-2acf-4459-b568-93a906d84008')
```
Comment (jfaracik) [inline cases/models.py]: Same as above (choice field vs hardcoding a row id?)

File: clc_queries/tests.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (jfaracik) [inline clc_queries/tests.py]: Delete if empty

File: clc_queries/views.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+from django.shortcuts import render
+
+# Create your views here.
```
Comment (jfaracik) [inline clc_queries/views.py]: Delete if empty

File: goods/views.py [hunk 8]
```diff
@@ -27,14 +32,32 @@ def get(self, request):
 
     def post(self, request):
         organisation = get_organisation_by_user(request.user)
-
         data = JSONParser().parse(request)
         data['organisation'] = organisation.id
         data['status'] = GoodStatus.DRAFT
+        # if data['not_sure_details_control_code'] is not None:
```
Comment (jfaracik) [inline goods/views.py]: Commented out code

File: goods/views.py [hunk 9]
```diff
@@ -27,14 +32,32 @@ def get(self, request):
...
+        #     data['control_code'] = data['not_sure_details_control_code']
         serializer = GoodSerializer(data=data)
 
         if serializer.is_valid():
-            serializer.save()
+            if not data['validate_only']:
+                good = serializer.save()
+                if data['is_good_controlled'] == GoodControlled.UNSURE:
+                    # automatically raise a CLC query case
+
```
Comment (jfaracik) [inline goods/views.py]: spacing

File: clc_queries/views.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+    def put(self, request, pk):
+        """
+        Update a clc query instance.
+        """
+        with reversion.create_revision():
+            serializer = ClcQueryUpdateSerializer(get_clc_query_by_pk(pk), data=request.data, partial=True)
+
+            if serializer.is_valid():
+                reversion.set_comment("Updated clc query details")
```
Comment (lukesa) [inline clc_queries/views.py]: should it come from content repo?

File: cases/views.py [hunk 11]
```diff
@@ -94,8 +94,15 @@ class ActivityList(APIView):
 
     def get(self, request, pk):
         case = get_case(pk)
+        print(dir(case))
```
Comment (andytaylorkainos) [inline cases/views.py]: remove print statements please

File: case_types/tests.py (hunk 2 above)
Comment (antonhughes) [reply case_types/tests.py]: removed

**Pull request merged**